src/hotspot/share/opto/gcm.cpp

```diff
@@ -644,60 +744,55 @@ Block* PhaseCFG::insert_anti_dependences(Block* LCA, Node* load, bool verify) {
   // The relevant stores "nearby" the load consist of a tree rooted
   // at initial_mem, with internal nodes of type MergeMem.
   // Therefore, the branches visited by the worklist are of this form:
-  //    initial_mem -> (MergeMem ->)* store
+  //    initial_mem -> (MergeMem ->)* Memory state modifying node
+  // Memory state modifying nodes include Store and Phi nodes and any node for which needs_anti_dependence_check()
+  // returns true.
```

Comment:
Basically this is the idea:
`needs_anti_dependence_check() == true` for Loads
`needs_anti_dependence_check() == false` for MergeMem, Phi, Store

So your `returns true` is not correct, right? Should it not be `returns false`?